lib/internal/source_map/source_map_cache.js

```diff
@@ -136,17 +152,86 @@ function sourceMapCacheToObject() {
 
 // Since WeakMap can't be iterated over, we use Module._cache's
 // keys to facilitate Source Map serialization.
+//
+// TODO(bcoe): this means we don't currently serialize source-maps attached
+// to error instances, only module instances.
 function appendCJSCache(obj) {
   const { Module } = require('internal/modules/cjs/loader');
   Object.keys(Module._cache).forEach((key) => {
     const value = cjsSourceMapCache.get(Module._cache[key]);
     if (value) {
-      obj[`file://${key}`] = value;
+      obj[`file://${key}`] = {
+        url: value.url,
+        data: value.data
+      };
+    }
+  });
+}
+
+// Create a prettified stacktrace, inserting context from source maps
+// if possible.
+const prepareStackTrace = (globalThis, error, trace) => {
+  const { SourceMap } = require('internal/source_map/source_map');
+  const ErrorToString = Error.prototype.toString;
```

Comment:
this line needs to be hoisted out, its purposely saving the original value of `Error.prototype.toString`.